core/spec.md

```diff
@@ -2031,45 +2070,31 @@ new attributes within the model definitions themselves), but only if
 the server supports them. Servers MUST reject model definitions that include
 unknown model language attributes.
 
-Once a Registry has been created, implementations MAY choose to limit the
-types of changes made to the model - for example, to ensure backwards
-compatibility of clients or to ensure existing entities do not need to be
-changed to be consistent with the new model.
-
-Implementations are REQUIRED to ensure that after any model changes are made,
-all of the entities in the Registry are valid with respect to the new model
-definition (including all Versions of Resources). How this is achieved will
-be implementation specific. For example, implementations can choose to
-automatically modify existing entities, or even to delete non-conforming
-entities (such as when Groups or Resource types are removed). However, it is
-STRONGLY RECOMMENDED that implementations not delete entities due to attribute
-modifications.
-
-When model changes are made, it might be necessary to modify existing entity's
-attributes to ensure they are conformant with the new model. The following
-ordered suggestions are defined to provide consistency across implementations,
-but it is NOT REQUIRED for implementations to follow these rules if a
-different set of changes are more appropriate:
-- If a `default` value is defined for the attribute, then it SHOULD be used.
-- If the attribute is OPTIONAL, then the attribute SHOULD be deleted.
-- If an enum is defined and `strict` is `true`, the first `enum` value SHOULD
-  be used.
-- If valid, the zero value for the attribute's type SHOULD be used:
-  - Array, Map, Object: empty value (e.g. `[]`, `{}`, `{}` respectively).
-  - Boolean: false.
-  - Numeric: zero.
-  - String, URI, URIReference, URI-Template, URL: empty string (`""`).
-  - Timestamp: zero (00:00:00 UTC Jan 1, 1970).
-- If valid, String attributes SHOULD use `"undefined"`.
-
-Changes to the model MUST NOT include changing the `hasDocument` aspect of a
-Resource from `false` to `true` if there are attributes defined on that
-Resource whose names conflict with the `RESOURCE*` attributes. Any attempt do
-to so MUST generate an error.
-
-If a backwards incompatible change is needed, and the existing entities need
-to be preserved, then it is RECOMMENDED that users define a new Group or
-Resource types for future instances of those entities.
+Once a Registry has been created, changes to the model are limited to:
+- Adding new Groups or Resources.
+- Adding new OPTIONAL attributes.
+
+Any model change that might result in existing data within the Registry
+becoming non-compliant MUST generate an error, even if it does not do so. For
```

Comment:
What is the intent of the following?

>, even if it does not do so.

Reply:
even if the data doesn't become invalidated because they got lucky "this time", doesn't mean it's an ok change to make. I'm not happy with that phrasing I used, so I'm open to an alternative.

Reply:
e.g. changing an optional attr from string to int would be safe if that attr isn't used yet. But it's not a valid change to make in general.

Reply:
How about:
> , even if the existing data doesn't result in non-compliance.